### ISSUE TITLE
Add "Hue-Shift" masking to Xiexe Toon

### ISSUE THREAD

**Issue by 3x1t-5tyl3** [Neos-Metaverse/NeosPublic#2028]
## Is your feature request related to a problem? Please describe.
It would allow to hue shift color one part differently than others..

## Relevant issues
#1951 #1929 #1473 #1928 #1791 

## Describe the solution you'd like
Add a black-white texture mask as similar issues above which would:
Black: No masking. The texture takes control over the color.
White: Takes full control over the hue.
Anything in-between would be obviously be inbetween. See video below.
This would apply to Emission and Albedo textures.

## Describe alternatives you've considered
Material stacking though that is very limited and costs another render pass (or multiple depending on the shader used)

## Additional context
https://user-images.githubusercontent.com/61988002/114921731-2cf57c00-9e2b-11eb-93f2-c674999c4a4b.mp4

Initial request by: @epicEaston197

**Comment by epicEaston197** [comment]
Xiexe Toon does not have a Hue-Shift slider to begin with Hue-Shifting is not a property that the material has at the moment

**Comment by 3x1t-5tyl3** [comment]
> Xiexe Toon does not have a Hue-Shift slider to begin with Hue-Shifting is not a property that the material has at the moment so on top of having a slider to do this we need to have a mask for it to do this
The video above is literally XiexeToon in unity.

**Comment by epicEaston197** [comment]
> > Xiexe Toon does not have a Hue-Shift slider to begin with Hue-Shifting is not a property that the material has at the moment so on top of having a slider to do this we need to have a mask for it to do this
> 
> The video above is literally XiexeToon in unity. Besides you were the one initially asking for it lmao?

yeah I know I wasn't saying I was against it I was saying that we need a hue slider

**Comment by 3x1t-5tyl3** [comment]
It was poyomi as I wasn't able to load Xiexe into unity and got pissed off 2 minutes into it. Especially while you were on my neck in-world about it.

**Comment by epicEaston197** [comment]
oh yeah sorry about that

**Comment by shininghero** [comment]
Looks like the latest version of Xiexe's supports hue shifting now.
![image](https://user-images.githubusercontent.com/17071843/148814939-dfee024c-68cf-4745-a629-820b323434e1.png)
